python_modules/dagster/dagster/core/definitions/repository_definition.py

```diff
@@ -1232,5 +1248,12 @@ def _process_and_validate_target(
         pipelines_or_jobs[target.name] = target
 
 
+def _validate_asset_resources(
+    assets_defs: Sequence["AssetsDefinition"], source_assets: Sequence[SourceAsset]
+):
+    """Validate that all asset resource requirements are satisfied and that no two assets have
+    conflicting definitions for the same resource"""
```

Comment:
@dpeng817 is there some place in the codebase where similar logic is implemented that I could steal from?